src/main/webapp/scripts/personalised-recommendations.js

```diff
@@ -12,67 +12,34 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-const topRecommendations = [
-  {
-    name: "Pret A Manger",
-    latitude: 51.5252574,
-    longitude: -0.1282885,
-    description: "Counter-serve chain for ready-made sandwiches plus breakfast, coffee, soups & salads.",
-    crowd: 3,
-    price: 1
-  },
-  {
-    name: "Wasabi",
-    description: "Light-filled, modern chain branch serving up bento boxes, sushi and hot Japanese dishes.",
-    latitude: 51.5172219,
-    longitude: -0.1334257,
-    crowd: 3,
-    price: 2
-  },
-  {
-    name: "Shake Shack",
-    description: "Branch of an American chain known for its gourmet burgers & frozen custard shakes. Dog-friendly.",
-    latitude: 51.5172383,
-    longitude: -0.1399918,
-    crowd: 2,
-    price: 2
-  },
-  {
-    name: "Taco Bell",
-    latitude: 51.5203205,
-    longitude: -0.125156,
-    description: "Fast-food chain serving Mexican-inspired fare such as tacos, quesadillas & nachos.",
-    crowd: 4,
-    price: 1
-  },
-  {
-    name: "KFC",
-    latitude: 51.5204878,
-    longitude: -0.1431986,
-    description: "Fast-food chain known for its buckets of fried chicken, plus wings & sides.",
-    crowd: 4,
-    price: 2
-  }
-];
-
-/* Load top 5 recommendations */
 function loadTopRecommendations() {
+  // Get category, costRating and crowdRating from user input radio buttons
+  const chosenCategory = document.querySelector('input[name="recommendation-category"]:checked').value;
+  const preferredCost = document.querySelector('input[name="price"]:checked').value;
+  const preferredCrowd = document.querySelector('input[name="crowd"]:checked').value;
+  const url = "recommender?category=" + chosenCategory +"&cost-rating=" 
+      + preferredCost +"&crowd-rating=" + preferredCrowd;
+  fetch(url).then(response => response.json()).then((recommendations) => {
+    displayRecommendation(recommendations);
+  });
+}
+
+/* Update HTML to display recommendation */
+function displayRecommendation(recommendations) {
   const topRecommendationsList = document.getElementById("top-recommendations-list");
   topRecommendationsList.innerHTML = "";
-
-  const recommendationBox = document.createElement("div");
-  recommendationBox.className = "recommendation-box";
-  for (var i = 1; i <= topRecommendations.length; i++) {
-    const topRecommendation = topRecommendations[i - 1];
+  for (var i = 0; i < 5; i++) {
```

Comment:
Same here, what if we received less than 5 recommendations from the server?

Reply:
Thanks! I'll add some logic to cover this case.